engine/launcher/src/main/scala/org/enso/launcher/upgrade/LauncherUpgrader.scala

```diff
@@ -217,28 +266,53 @@ class LauncherUpgrader(
 
   @scala.annotation.tailrec
   private def nextVersionToUpgradeTo(
-    release: LauncherRelease,
+    currentTargetRelease: LauncherRelease,
     availableVersions: Seq[SemVer]
   ): LauncherRelease = {
-    val recentEnoughVersions =
-      availableVersions.filter(
-        _.isGreaterThanOrEqual(release.minimumVersionToPerformUpgrade)
+    assert(
+      currentTargetRelease.minimumVersionToPerformUpgrade.isGreaterThan(
+        CurrentVersion.version
       )
+    )
+
+    // We look at older versions that are satisfying the minimum version
+    // required to upgrade to currentTargetRelease.
+    val recentEnoughVersions =
+      availableVersions.filter { possibleVersion =>
+        val canUpgradeToTarget = possibleVersion.isGreaterThanOrEqual(
+          currentTargetRelease.minimumVersionToPerformUpgrade
+        )
+        val isEarlierThanTarget =
+          possibleVersion.isLessThan(currentTargetRelease.version)
+        canUpgradeToTarget && isEarlierThanTarget
+      }
+
+    // We take the oldest of these, hoping that it will yield the shortest
+    // upgrade path (perhaps it will be possible to upgrade directly from
+    // current version)
     val minimumValidVersion = recentEnoughVersions.sorted.headOption.getOrElse {
       throw UpgradeError(
-        s"Upgrade failed: To continue upgrade, a version at least " +
-        s"${release.minimumVersionToPerformUpgrade} is required, but no " +
-        s"valid version satisfying this requirement could be found."
+        s"Upgrade failed: To continue upgrade, at least version " +
+        s"${currentTargetRelease.minimumVersionToPerformUpgrade} is required, " +
+        s"but no upgrade path has been found from the current version " +
+        s"${CurrentVersion.version}. " +
+        s"Please manually download a newer release from " +
+        s"${LauncherRepository.websiteUrl}"
```

Comment:
Added a note that links to the GH repository where the user can manually download a new release.